src/sync/sync_session.cpp

```diff
@@ -473,18 +474,22 @@ void SyncSession::handle_progress_update(uint64_t downloaded, uint64_t downloada
     }
 }
 
-void SyncSession::NotifierPackage::update(const Progress& current_progress)
+void SyncSession::NotifierPackage::update(const Progress& current_progress, bool data_is_fresh)
 {
-    if (is_streaming || captured_transferrable)
+    if (is_streaming || captured_transferrable || !data_is_fresh)
         return;
 
     captured_transferrable = direction == NotifierType::download ? current_progress.downloadable
                                                                  : current_progress.uploadable;
 }
 
-std::function<void()> SyncSession::NotifierPackage::create_invocation(const Progress& current_progress, bool& is_expired) const
+std::function<void()> SyncSession::NotifierPackage::create_invocation(const Progress& current_progress,
+                                                                      bool& is_expired) const
 {
-    REALM_ASSERT(is_streaming || captured_transferrable);
+    // It's possible for a non-streaming notifier to not yet have fresh transferrable bytes data.
+    // In that case, we don't call it at all.
```

Comment:
Oh, I see.

Every time `create_invocation()` is called, it's preceded by a call to `update()`. `update()` is responsible for checking if the newest data is fresh or not, and then updating the `captured_transferrable` value. So the first time fresh data is sent down `captured_transferrable` is set to a non-`none` value, and the block should be called.

That being said, the fact that I had to explain the above means that the code needs to be refactored for clarity, so let me think about how best to move things around and make my intent explicit...